### PR TITLE
Feature/PROD-1749 | Update api docs to reflect new msdp OTA workflow

### PR DESCRIPTION
**Pull request by sunny-trip-ninja**
## LocalQA approved by
@Sanpele 

## Description
- Updated the MSDP search documentation including
  - /msdp/detail/hotel
  - /pre-booking
  - /ota-booking
  - /ticketing
  - /cancel

## How to Test
1. Checkout this branch
2. Run  `npx @redocly/openapi-cli preview-docs tn_api_msdp_v1.yml`
3. Go to http://127.0.0.1:8080
4. You should see the documentation for the mentioned endpoints above in `Description`

### CODE REVIEW

**Comment by mschembri-TN** [comment]
Task linked: [PROD-1749 MSDP | OTA Booking | Create Docs for /pre-booking, /booking, /ticketing, /cancel](https://app.clickup.com/t/14197839/PROD-1749)

**Comment by mschembri-TN** [comment]
Task linked: [PROD-1292 Update API Docs to reflect new MSDP Search functionality](https://app.clickup.com/t/14197839/PROD-1292)

File: tn_api_msdp_v1.yml [hunk 1]
```diff
@@ -596,6 +779,151 @@ components:
...
+                required:
+                    - msdp_package_id
+                    - trip_id
+                    - segment_ids
+                    - hotel_room_details
+                properties:
+                    msdp_package_id:
+                        type: string
+                        example: "6be6e72c60102bd489ac46434d4ac6c11e047ee8"
+                        description: MSDP Pacakge ID returned in msdp search response
```
Comment (vickchungg) [inline tn_api_msdp_v1.yml]: Typo in Package

File: tn_api_msdp_v1.yml [hunk 2]
```diff
@@ -596,6 +779,151 @@ components:
...
+                type: object
+                required:
+                    - msdp_package_id
+                    - trip_id
+                    - segment_ids
+                properties:
+                    msdp_package_id:
+                        type: string
+                        example: "6be6e72c60102bd489ac46434d4ac6c11e047ee8"
+                        description: MSDP Pacakge ID returned in msdp search response
```
Comment (vickchungg) [inline tn_api_msdp_v1.yml]: Typo here too

File: tn_api_msdp_v1.yml [hunk 3]
```diff
@@ -596,6 +779,151 @@ components:
...
```
Comment (vickchungg) [inline tn_api_msdp_v1.yml]: Typo here

**Comment by Sanpele** [comment]
Some notes:

Small stuffs:
1. The hotel_location field is missing in the MSDP Get Hotel Details request. Should be added and maybe as required, as is required for webbeds and required to return the lat/lng for a hotel. 
2. For the MSDP Get Hotel Details response description object, the address, rooms and meals fields don't have a description.
Screenshot 2023-01-03 at 10 41 33 AM
3. for the MSDP Get Hotel Details resposne, the hotel_description, search_request and detail_request fields are missing. Not sure if the search/detail request need to be in the docs but the description should be prob.
4. for the Price Confirmation Report the hotel_room_details fields are not tagged as required, but I believe they should be.
Screenshot 2023-01-03 at 10 46 00 AM
5. for the Booking Report and Ticketing Report the type of the hotel_trace_id's array should be array of strings not array of arrays.
Screenshot 2023-01-03 at 10 57 44 AM
6. The http method of cancel booking should be delete not post
Screenshot 2023-01-03 at 10 58 38 AM
Screenshot 2023-01-03 at 10 58 42 AM

and maybe this should be msdp/cancel
Screenshot 2023-01-03 at 10 58 54 AM
Screenshot 2023-01-03 at 10 59 33 AM

Out of Scope:

It would be good for us at some point to include a flow diagram for MSDP for each of the QT/OTA paths. We do have one for Flight search for QT, I believe it would be helpful to future users to have a more visual idea of how the requests interact together. Plus I think especially for the OTA flow it would be useful have a bit more visual of a display, as I think its a bit less intuitive. 
* One way to maybe do this would be to separate the QT/OTA endpoints into their own locations, whether in diff files or whether we could do this somehow in the MSDP page.

**Comment by sunny-trip-ninja** [comment]
> Some notes:
> 
> Small stuffs:
> 
> 1. The hotel_location field is missing in the MSDP Get Hotel Details request. Should be added and maybe as required, as is required for webbeds and required to return the lat/lng for a hotel.
> 2. For the MSDP Get Hotel Details response description object, the address, rooms and meals fields don't have a description.
>    Screenshot 2023-01-03 at 10 41 33 AM
> 3. for the MSDP Get Hotel Details resposne, the hotel_description, search_request and detail_request fields are missing. Not sure if the search/detail request need to be in the docs but the description should be prob.
> 4. for the Price Confirmation Report the hotel_room_details fields are not tagged as required, but I believe they should be.
>    Screenshot 2023-01-03 at 10 46 00 AM
> 5. for the Booking Report and Ticketing Report the type of the hotel_trace_id's array should be array of strings not array of arrays.
>    Screenshot 2023-01-03 at 10 57 44 AM
> 6. The http method of cancel booking should be delete not post
>    Screenshot 2023-01-03 at 10 58 38 AM
>    Screenshot 2023-01-03 at 10 58 42 AM
> 
> and maybe this should be msdp/cancel Screenshot 2023-01-03 at 10 58 54 AM Screenshot 2023-01-03 at 10 59 33 AM
> 
> Out of Scope:
> 
> It would be good for us at some point to include a flow diagram for MSDP for each of the QT/OTA paths. We do have one for Flight search for QT, I believe it would be helpful to future users to have a more visual idea of how the requests interact together. Plus I think especially for the OTA flow it would be useful have a bit more visual of a display, as I think its a bit less intuitive.
> 
> * One way to maybe do this would be to separate the QT/OTA endpoints into their own locations, whether in diff files or whether we could do this somehow in the MSDP page.



@Sanpele Thanks for reviewing the code, really appreciate the effort!

1-5 is now addressed
6) The endpoint method is post for OTA cancel flow

For the out of scope stuff, would you please create a ticket and we'll address it at a later time. Thanks!

**Comment by Sanpele** [comment]
Task for doc flow improvement : https://app.clickup.com/t/860pf3qgc

**Comment by Sanpele** [comment]
just noticed another small thing. For endpoints with longer names, if i navigated to them the URL would have %20 values for spaces between words. e.g. for get hotel details
Screenshot 2023-01-05 at 10 05 11 AM
It might be better to remove the spaces, is def more readable. You can do this by removing the spaces from the operationId from that endpoint. 

This is happening for: 
Get Flight Results
Get Hotel Results
Remove Hotel Result
Get Hotel Details

One other thing, it might make sense to either prefix all the endpoints in the sidebar with MSDP, or none. RN only half have. 
e.g. 
MSDP Price Confirm
MSDP Booking Report
...
Wat u think

Screenshot 2023-01-05 at 10 10 25 AM

**Comment by sunny-trip-ninja** [comment]
> just noticed another small thing. For endpoints with longer names, if i navigated to them the URL would have %20 values for spaces between words. e.g. for get hotel details Screenshot 2023-01-05 at 10 05 11 AM It might be better to remove the spaces, is def more readable. You can do this by removing the spaces from the operationId from that endpoint.
> 
> This is happening for: Get Flight Results Get Hotel Results Remove Hotel Result Get Hotel Details
> 
> One other thing, it might make sense to either prefix all the endpoints in the sidebar with MSDP, or none. RN only half have. e.g. MSDP Price Confirm MSDP Booking Report ... Wat u think
> 
> Screenshot 2023-01-05 at 10 10 25 AM

Updated!

**Comment by Sanpele** [comment]
everything looks good to me, one thing tho. I think maybe ota-booking and QT cancel may be missing? although maybe not. I do know booking/ and ota-booking/ have diff endpoints and only booking/ is present in the docs. Also 
we have a cancel endpoint and a delete HTTP method on the booking/ endpoint but only have the cancel included in the docs. they missing or left out on purpose or something?

**Comment by sunny-trip-ninja** [comment]
> everything looks good to me, one thing tho. I think maybe ota-booking and QT cancel may be missing? although maybe not. I do know booking/ and ota-booking/ have diff endpoints and only booking/ is present in the docs. Also we have a cancel endpoint and a delete HTTP method on the booking/ endpoint but only have the cancel included in the docs. they missing or left out on purpose or something?

All these are based on the OTA workflow, theres no QT related component for this docs. Could add QT related docs in the future

**Comment by Sanpele** [comment]
@sunny-trip-ninja 
one small thing, is this the right header and title in the side bar? seems like it it should be pre-booking not price confirmation to me although for OTA maybe not.
Screenshot 2023-01-24 at 4 42 47 PM

**Comment by sunny-trip-ninja** [comment]
> @sunny-trip-ninja one small thing, is this the right header and title in the side bar? seems like it it should be pre-booking not price confirmation to me although for OTA maybe not. Screenshot 2023-01-24 at 4 42 47 PM

hmmm i copied what we have currently for the normal OTA flow

**Review by Sanpele — APPROVED** [review]

**Comment by Sanpele** [comment]
localQA approved, looks good. All changes present and viewable and everything seems to be working.

**Pull request merged**